packages/itwin/tree-widget/src/components/trees/TreeWithRuleset.tsx

```diff
@@ -114,30 +113,38 @@ export const ControlledTreeWrapper: React.FC<ControlledTreeProps> = (
   const { nodeLoader } = usePresentationTreeNodeLoader({
     imodel: props.iModel,
     ruleset: props.rulesetId,
-    preloadingEnabled: false,
     pagingSize: props.pageSize || 20,
-    dataProvider: props.dataProvider,
   });
-  nodeLoader.dataProvider;
+
   const modelSource = nodeLoader.modelSource;
   const unifiedSelectionEventHandler = useUnifiedSelectionTreeEventHandler({
     nodeLoader,
     collapsedChildrenDisposalEnabled: true,
   });
 
-  const visibleNodes = useVisibleTreeNodes(modelSource);
+  const [height, setHeight] = useState(0);
+  const [width, setWidth] = useState(0);
+  const handleResize = useCallback((w: number, h: number) => {
+    setHeight(h);
+    setWidth(w);
+  }, []);
+  const ref = useResizeObserver<HTMLDivElement>(handleResize);
+
+  const treeModel = useTreeModel(modelSource);
 
   return (
-    <ControlledTree
-      visibleNodes={visibleNodes}
-      nodeLoader={nodeLoader}
-      treeEvents={unifiedSelectionEventHandler}
-      selectionMode={SelectionMode.Extended}
-    />
+    <div ref={ref} style={{ width: "100%", height: "100%" }}>
```

Comment:
Why was this update needed?

Reply:
I see this update all over. I assume it's to make the component more adaptive?

Reply:
one of the breaking changes early on, https://github.com/iTwin/itwinjs-core/blob/master/docs/changehistory/NextVersion.md#controlledtree-api-changes